libs/init/migrate/src/mill/init/migrate/PackageTree.scala

```diff
@@ -0,0 +1,90 @@
+package mill.init.migrate
+
+import mill.constants.CodeGenConstants.{nestedBuildFileNames, rootBuildFileNames}
+
+import java.io.PrintStream
+import scala.collection.mutable
+import scala.util.Using
+
+opaque type PackageTree = Tree[PackageRepr]
```

Comment:
Let's just use a normal `type` alias or `case class` here

Reply:
In general, we aim to use simpler language features over more powerful ones, unless there is a strong reason otherwise. In this case, there is no strong reason to use an `opaque type`, so we use `case class` which people are more familiar with